atd-vzv/src/views/summary/CrashesByMode.js

```diff
@@ -174,6 +174,21 @@ const CrashesByMode = () => {
     datasets: !!chartData && createTypeDatasets(),
   };
 
+  // Get an array of annual totals for the selected crash type
+  const getYearTotalsArray = useMemo(() => {
+    const yearTotalsArray = yearsArray().map((year, index) => {
+      let currentYearTotal = 0;
+      data.datasets &&
```

Comment:
on first render `data.dataset` is null so this will fail unless i do this check

Reply:
thanks @roseeichelmann for making this change! this looks great.

Worth noting that because `data` is not memoized, this hook is still going to run more than it needs to, but that problem feels out of scope.

Reply:
+1 - a small improvement could be to update the dependency array with `data` and throw a `if (!data?.datasets) return;` at the beginning of the function passed to useMemo to skip the rest of the code when there is no dataset. Not a huge improvement, and I think we can punt until we pay off the debt here.